client/dashboard/app/router/emails.tsx

```diff
@@ -104,6 +107,35 @@ export const chooseEmailSolutionRoute = createRoute( {
 	)
 );
 
+export const addProfessionalEmailRoute = createRoute( {
+	head: () => ( {
+		meta: [
+			{
+				title: __( 'Add Professional Email' ),
+			},
+		],
+	} ),
+	getParentRoute: () => rootRoute,
+	path: 'emails/add-professional-email/$domain',
+	loader: async ( { params: { domain: domainName } } ) => {
+		const products = queryClient.ensureQueryData( productsQuery() );
+		const site = queryClient.ensureQueryData( siteBySlugQuery( domainName ) );
+
+		const domain = await queryClient.ensureQueryData( domainQuery( domainName ) );
+		const mailboxAccounts = await queryClient.ensureQueryData(
+			mailboxAccountsQuery( domain?.blog_id, domainName )
```

Comment:
`blog_id` isn't nullable, we can just use `domain.blog_id` instead.

Reply:
Addressed on https://github.com/Automattic/wp-calypso/pull/106524/commits/785c667fc2d5f57bbf95569a1a2034d9457f7eae